include/gpio_pin.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "gpio_wrapper.hpp"
+#include "stdint.h"
+
+namespace edge {
+namespace drivers {
+
+enum GPIODirection { IN, OUT };
+
+class GPIOPin {
+public:
+    GPIOPin(uint32_t const pin_number, GPIODirection const pin_direction) :
```

Comment:
why uint32_t const

Is this a stylistic thing? fine if so, just curious

Reply:
east const? yea thats style, we can swap it

Reply:
also making constructor args const, idc tho. was just curious

Reply:
force of habit